app/models/manageiq/providers/openstack/inventory/persister/definitions/storage_collections.rb

```diff
@@ -1,39 +1,34 @@
 module ManageIQ::Providers::Openstack::Inventory::Persister::Definitions::StorageCollections
   extend ActiveSupport::Concern
 
-  def initialize_storage_inventory_collections
-    %i(cloud_volumes
-       cloud_volume_snapshots
-       cloud_volume_types
-       ).each do |name|
-
-      add_collection(storage, name) do |builder|
-        builder.add_properties(:model_class => "#{cinder_manager.class}::#{name.to_s.classify}".constantize)
-        builder.add_properties(:parent => manager.cinder_manager) if targeted?
-        builder.add_default_values(:ems_id => cinder_manager.id)
-      end
-    end
-    add_cloud_volume_backups
-  end
-
-  def add_cloud_volume_backups(extra_properties = {})
-    add_collection(storage, :cloud_volume_backups, extra_properties) do |builder|
-      builder.add_properties(:model_class => "#{cinder_manager.class}::CloudVolumeBackup".constantize)
-      builder.add_properties(:parent => manager.cinder_manager) if targeted?
-      builder.add_default_values(:ems_id => cinder_manager.id)
-
+  def initialize_cinder_inventory_collections
+    add_cinder_collection(:cloud_volumes)
+    add_cinder_collection(:cloud_volume_snapshots)
+    add_cinder_collection(:cloud_volume_types)
+    add_cinder_collection(:cloud_volume_backups) do |builder|
       # targeted refresh workaround-- always refresh the whole backup collection
       # regardless of whether this is a TargetCollection or not
       # because OpenStack doesn't give us UUIDs of changed volume_backups,
       # we just get an event that one of them changed
-      if references(:cloud_volume_backups).present?
-        builder.add_properties(:targeted => false)
-      end
+      builder.add_properties(:targeted => false) if references(:cloud_volume_backups).present?
+    end
+  end
+
+  def initialize_swift_inventory_collections
+    add_swift_collection(:cloud_object_store_objects)
+    add_swift_collection(:cloud_object_store_containers)
+    add_cloud_collection(:cloud_tenants) do |builder|
+      builder.add_properties(:strategy => :local_db_cache_all, :complete => false)
     end
   end
 
   def add_cinder_collection(collection_name, extra_properties = {}, settings = {}, &block)
     settings[:parent] ||= cinder_manager
     add_collection(storage, collection_name, extra_properties, settings, &block)
   end
+
+  def add_swift_collection(collection_name, extra_properties = {}, settings = {}, &block)
+    settings[:parent] ||= swift_manager
+    add_collection(storage, collection_name, extra_properties, settings, &block)
+  end
```

Comment:
These are intended to be similar to the `add_storage_collection` from core, but since openstack has two storage managers we have to be more specific.